### DISCUSSION TITLE
Architecture/notes of OT implementation

### DISCUSSION THREAD

**Discussion by Jatsekku** [openthread/openthread#8577] [Q&A] [answered]
Hi,

I would like to better understand how OT works under the hood. 
Are there any documents about implementation itself?
Or could you provide me any hints how to start analyzing such a complex code base?

**Comment by jwhui** [comment, marked as answer]
See the [Thread Specification](https://www.threadgroup.org/support#specifications) for the Thread protocol itself.

As for OpenThread-specific documentation, you can find all resources at https://openthread.io/ and here on GitHub.

**Reply by abtink** [reply]
@Jatsekku This YouTube [video](https://youtu.be/Zlo0V55gJlk) (from  Thread Member Meeting in April 2022) may also help. It provides a very high-level overview of OpenThread stack.
